game/src/test/scala/io/aigar/game/ResourcesSpec.scala

```diff
@@ -66,4 +88,31 @@ class ResourcesSpec extends FlatSpec with Matchers {
 
     cell.mass should equal(30)
   }
+
+  it should "returns a list of scoreMessages from multiple positions and players" in {
+    val resource = new ResourceType(new Grid(0, 0), 0, 0, 5, 10)
+    val r1 = Vector2(10f, 10f)
+    val r2 = Vector2(50f, 50f)
+    resource.positions = List(r1, r2)
+
+    val p1 = new Player(1, Vector2(10f, 10f))
+    val p2 = new Player(2, Vector2(50f, 50f))
+
+    val resourceMessages = resource.detectCollisions(List(p1, p2))
+
+    resourceMessages should contain only (ScoreMessage(p1.id, 10), ScoreMessage(p2.id, 10))
+  }
+
+  it should "returns an empty list of scoreMessages" in {
```

Comment:
Similar here, could we say that this is when no collision occurs?
E.g.
`return an empty list of ScoreMessages when no collision occurs`